provider/azure/internal/ad/serviceprincipals.go

```diff
@@ -110,9 +111,115 @@ func (client ServicePrincipalsClient) ListResponder(resp *http.Response) (result
 	err = autorest.Respond(
 		resp,
 		client.ByInspecting(),
-		azure.WithErrorUnlessStatusCode(http.StatusOK),
+		WithOdataErrorUnlessStatusCode(http.StatusOK),
+		autorest.ByUnmarshallingJSON(&result),
+		autorest.ByClosing())
+	result.Response = autorest.Response{Response: resp}
+	return
+}
+
+func (client ServicePrincipalsClient) ListPasswordCredentials(objectId string) (result PasswordCredentialsListResult, err error) {
```

Comment:
This code will be replaced as soon as the Azure SDK is updated. The SDK is auto-generated, and its comments are pretty awful, so I'm not going to spend time on something that'll become irrelevant anyway.